torch/csrc/jit/passes/freeze_module.cpp

```diff
@@ -227,6 +227,13 @@ class AttributePropagator {
         elems.set(i, overrideGradient(elems.extract(i)));
       }
       attr = std::move(elems);
+    } else if (attr.isGenericDict()) {
```

Comment:
For the override gradient implementation here, can you call `getSubValues` and then remove the gradient for all tensor subvalues ?

Reply:
the gradient  is not overridden inplace because it is detached so that the original module has no side effects so I can really use getSubValues

Reply:
Okay, that makes sense. If that's the case don't you need to return the recursively constructed new attribute and reset it as the attribute ? 

here for example we're not returning a new tuple with the detatched gradients. so it's either inplace, which we're not trying to do, or we're  not changing anything. https://github.com/pytorch/pytorch/blob/c34ee73af4702a79d024bd287385e944648be6fb/torch/csrc/jit/passes/freeze_module.cpp#L223

Reply:
cc @Krovatkin @smessmer @zdevito how to do correct detaching of gradients in these cases